### ISSUE TITLE
Add Unit Type to Support Void Methods and requests for non-generic Result types

### ISSUE THREAD

**Issue by ardalis** [ardalis/Result#80]
MediatR has the same problems with returning things from void methods, and they've solved for years by using a Unit type:
https://github.com/jbogard/MediatR/blob/master/src/MediatR.Contracts/Unit.cs

Should we use a similar type included in this library for this purpose, and document its usage?

See related issues:

- #77 
- #65 
- #41 

What do you think? Comment or add an appropriate emoji.

**Comment by Artem-Romanenia** [comment]
Have you considered just having IResult as
```csharp
public interface IResult
{
    ResultStatus Status { get; }
    IEnumerable<string> Errors { get; }
    List<ValidationError> ValidationErrors { get; }
}

public interface IResult<T> : IResult
{
    T GetValue();
}
```
and then having two implementations - `Result` and `Result<T>`, with `Result<T>.Success(T value)` and `Result.Success()`?

I understand that those two implementations would have to have a list of almost identical static methods, and also certain amount of other ugliness in different places, for example when converting Result to ActionResult, but maybe it`s worth the benefits?

I also understand the backward compatibility concerns, but were there any other reasons why you've decided against this in a first place?

**Comment by ardalis** [comment]
I think this is addressed by #89
